client/quantum_serverless/core/provider.py

```diff
@@ -292,26 +294,34 @@ def run(self, program: Program, arguments: Optional[Dict[str, Any]] = None) -> J
 
         return job_client.run(program, arguments)
 
+    def upload(self, program: Program):
+        """Uploads program."""
+        raise NotImplementedError
+
     def files(self) -> List[str]:
         """Returns list of available files produced by programs to download."""
         raise NotImplementedError
 
-    def download(self, file: str, download_location: str):
+    def file_download(self, file: str, download_location: str):
```

Comment:
I don't think anyone has implemented their own provider, so this probably is more academic than anything else, but for sake of completeness: should we deprecate the `download` function

Reply:
It is "experimental" feature :) 

but anyway it is a good practice. Deprecated `download` in https://github.com/Qiskit-Extensions/quantum-serverless/pull/1041/commits/6e690da4163ba58b8c453c55b0d5c3175d728dbc

Reply:
not, upload though. That is why I though it is a strange thing to deprecate half and not the other